cirq/google/calibration/workflow.py

```diff
@@ -337,19 +351,150 @@ def run_characterizations(
     return results
 
 
-@dataclasses.dataclass(frozen=True)
-class CircuitPhasedFSimCalibrationResults:
-    """Circuit-specific calibration results
+def zeta_chi_gamma_calibration_for_moments(
+    circuit_calibration: CircuitCalibration,
+    characterizations: List[PhasedFSimCalibrationResult],
+    gates_translator: Callable[[Gate], Optional[FSimGate]] = try_convert_sqrt_iswap_to_fsim,
+) -> CircuitCalibration:
+    """Compensates circuit against errors in zeta, chi and gamma angles.
+
+    This method creates a new circuit with a single-qubit Z gates added in a such way so that
+    zeta, chi and gamma angles discovered by characterizations are cancelled-out and set to 0.
+
+    This function preserves a moment structure of the circuit. All single qubit gates appear on new
+    moments in the final circuit.
+
+    Args:
+        circuit_calibration: Description of the circuit together with its calibration metadata that
+            is compatible with the characterizations argument.
+        characterizations: List of characterization results.
+        gates_translator: Function that translates a gate to a supported FSimGate which will undergo
+            characterization. Defaults to sqrt_iswap_gates_translator.
+
+    Returns:
+        Calibrated circuit together with its calibration metadata in CircuitCalibration object. The
```

Comment:
Why are we returning the CircuitCalibration instead of just the circuit? Is it to (indirectly) keep track of the mapping from the input circuit to the output circuit, namely after the new Z moments have been added?

Reply:
Added a comment: this is updated CircuitWithCalibration; the idea is that calibration can be composed. Will see how it ends.

Reply:
got it